core/src/main/java/org/apache/iceberg/AllManifestsTable.java

```diff
@@ -192,13 +191,11 @@ public List<DeleteFile> deletes() {
     @Override
     public CloseableIterable<StructLike> rows() {
       try (CloseableIterable<ManifestFile> manifests =
-          Avro.read(io.newInputFile(manifestListLocation))
-              .rename("manifest_file", GenericManifestFile.class.getName())
-              .rename("partitions", GenericPartitionFieldSummary.class.getName())
-              .rename("r508", GenericPartitionFieldSummary.class.getName())
+          InternalData.read(FileFormat.AVRO, io.newInputFile(manifestListLocation))
+              .setRootType(GenericManifestFile.class)
+              .setCustomType(508, GenericPartitionFieldSummary.class)
```

Comment:
Do we have a constant for this field id?

Reply:
The closes thing is probably to lookup the id from the schema:

```
ManifestFile.SCHEMA.findField("partitions.element").fieldId()
```

I'll use that since we don't have any static references to the field ids in these schemas.  This is clearly more readable.